### PR TITLE
Fix: avoid potential crash during certain verification paths.

### PR DESCRIPTION
**Pull request by bwindels**
`VerificationQREmojiOptions`, which was only used from `DeviceVerifyDialog`, wasn't ported to use QR code generation in the js-sdk, and could potentiall crash when clicking verify from either a key share dialog or the encrypted event dialog.

Instead of porting that legacy code, I removed it all together, and ported the usage of `DeviceVerifyDialog` to go over non-legacy code instead.

I wasn't able to test the key share dialog path as I have no idea how to make one pop up, but that path will disappear once we go full cross-signing as well, and I've read over my modifications several times and *should* work.

### CODE REVIEW

File: src/verification.js [hunk 1]
```diff
@@ -59,41 +59,82 @@ function UntrustedDeviceDialog(props) {
...
-                Modal.createTrackedDialog("Legacy verify session", "legacy verify session",
-                    ManualDeviceKeyVerificationDialog,
-                    {
-                        userId: user.userId,
-                        device,
-                    },
-                );
-            }
-        },
+    return new Promise(resolve => {
```
Comment (t3chguy) [inline src/verification.js:82]: this could be massively cleaned up by using 

```
const {finished} = Modal.create...
const [action] = await finished;
```

**Comment by bwindels** [comment]
Turns out that the E2E tests still relied on this dialog, and fixing them will be a bit of work, so postponing this PR and porting the legacy code over not to crash.

File: src/verification.js (hunk 1 above)
Comment (bwindels) [reply src/verification.js:82]: noted

Comment (bwindels) [reply src/verification.js:82]: The dialog gets closed before the verification is finished though...

Comment (t3chguy) [reply src/verification.js:82]: Right, I just mean in terms of flattening the code

**Pull request closed without merging**